nrf-hal-common/src/nvmc.rs

```diff
@@ -0,0 +1,101 @@
+//! HAL interface to the Non-Volatile Memory Controller (NVMC) peripheral.
+
+#[cfg(any(feature = "52840"))]
+use crate::pac::NVMC;
+#[cfg(any(feature = "9160"))]
+use crate::pac::NVMC_NS;
+use embedded_storage::nor_flash::{NorFlash, ReadNorFlash};
+
+/// Interface to an NVMC instance.
+pub struct Nvmc<'a, T: Instance, const N: usize> {
+    nvmc: T,
+    storage: &'a mut [u32; N],
```

Comment:
I'm not sure how you're planning on using this, but if you plan on offering non-blocking APIs we need to make sure this is `mem::forget`-safe (presumably by making it `'static`)

Also some rationale on why this is const-generic instead of a dynamic slice would be nice, and why this is using `u32` instead of `u8` as the element type – does the NVMC have alignment requirements?

Reply:
Yes, writes must be aligned to u32. I agree on the const generics though, a dynamic slice seems simpler.

Also, I'm not sure if there's UB issues with having a `&mut` to the data while you modify it / erase it through magic register writes...?

Reply:
> I'm not sure how you're planning on using this, but if you plan on offering non-blocking APIs we need to make sure this is `mem::forget`-safe (presumably by making it `'static`)

Commit 70aad9449bb83c0ad2fa3eb19f17044c270a6f9b makes its 'static.

Reply:
> Also some rationale on why this is const-generic instead of a dynamic slice would be nice.

Commit 70aad9449bb83c0ad2fa3eb19f17044c270a6f9b changes this to a dynamic slice. Dunno what I was thinking. :-)

Reply:
> Also, I'm not sure if there's UB issues with having a `&mut` to the data while you modify it / erase it through magic register writes...?

I don't fully understand the question. However, the data isn't actually modified via registers... the NVMC is put into various modes and then regular memory operations ensue.